libcontainer/cgroups/fscommon/utils.go

```diff
@@ -35,22 +35,42 @@ func ParseUint(s string, base, bitSize int) (uint64, error) {
 	return value, nil
 }
 
-// GetCgroupParamKeyValue parses a space-separated "name value" kind of cgroup
-// parameter and returns its components. For example, "io_service_bytes 1234"
-// will return as "io_service_bytes", 1234.
-func GetCgroupParamKeyValue(t string) (string, uint64, error) {
-	parts := strings.Fields(t)
-	switch len(parts) {
-	case 2:
-		value, err := ParseUint(parts[1], 10, 64)
-		if err != nil {
-			return "", 0, fmt.Errorf("unable to convert to uint64: %v", err)
-		}
+// ParseKeyValue parses a space-separated "name value" kind of cgroup
+// parameter and returns its key as a string, and its value as uint64
+// (ParseUint is used to convert the value). For example,
+// "io_service_bytes 1234" will be returned as "io_service_bytes", 1234.
+func ParseKeyValue(t string) (string, uint64, error) {
+	parts := strings.SplitN(t, " ", 3)
```

Comment:
SplitN should be called with 2, not 3?

Reply:
Specifying 3 instead of 2 here helps to catch a situation with an extra space (if there is one, we error out on `len(parts) != 2`.

Practically it does not matter much, as `ParseUint` will also fail given an extra a space, but I'd rather fail earlier.

Reply:
IOW
```go
parts := strings.SplitN(t, " ", 3)
if len(parts) != 2 {
   // return error
}
```

is a way to split t into two parts, while also checking there's no more than two parts.